src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -32,6 +32,7 @@ public static IEnumerable<object[]> Names_TestData()
             yield return new object[] { "name with spaces", "name with spaces" };
             yield return new object[] { "\uD800\uDC00", "\uD800\uDC00" };
             yield return new object[] { "\u043F\u0440\u0438\u0432\u0435\u0442", "\u043F\u0440\u0438\u0432\u0435\u0442" };
+            yield return new object[] { "🐵🐻#2faeea4e-c41b-4d38-85ea-124b4197280d-0", "🐵🐻#2faeea4e-c41b-4d38-85ea-124b4197280d-0" };
```

Comment:
If the relevant issue affects more than just assembly names, do we need to add tests elsewhere for other subsystems?

Reply:
I will add one for interop  utf8 string marshalling.

Reply:
Can you use the \u syntax instead of direct emoji characters? We try to do that since otherwise garbling can happen if someone changes the source code file encoding.

Reply:
Add another commit to use \u syntax instead.